exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -237,9 +238,8 @@ def measure_request_duration
         end
 
         def backoff?(retry_after: nil, retry_count:, reason:)
-          return false if retry_count > RETRY_COUNT
-
           @metrics_reporter.add_to_counter('otel.otlp_exporter.failure', labels: { 'reason' => reason })
```

Comment:
We were also not reporting the final failed attempt, thanks to @SomalianIvan for catching this.

Reply:
Any chance to add a test for this 🙏🏼 ?

Reply:
I remember why I didn't have a test for this originally. 

The backoff behaviour basically makes running tests really slow unless you stub backoff, which is where the metric reporting behaviour is.  
https://github.com/open-telemetry/opentelemetry-ruby/blob/469aa89ba180b1716a6a33c12647b32f1df4fed9/exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb#L239-L264

It's pretty awkward to test, if you have some ideas to simplify testing this I can circle back and add it.